src/components/activity-edit-form/ActivityEditForm.tsx

```diff
@@ -202,7 +175,7 @@ const ActivityEditForm = ({ onClose, activity }: ActivityEditFormProps) => {
                             className={classes.iconPicker}
                             label="Select an Icon"
                             iconKey={iconKey}
-                            onIconChange={handleIconChange}
+                            onIconChange={(iconKey) => setIcon(iconKey)}
```

Comment:
You should be able to ditch the arrow function in this case and simply pass `onIconChange={setIcon}`. If not then ignore my comment and leave it like that.